### ISSUE TITLE
"flutter run" on Linux fails with "undefined symbol: g_once_init_enter_pointer"

### ISSUE THREAD

**Issue by czaefferer** [flutter/flutter#145405]
### Steps to reproduce

I tried getting the package "flutter_secure_storage" to get to work in an app running on Linux, and for that I installed some packages required via brew. While doing that, brew updated all already installed packages, and after that, I cannot run any Flutter apps on Linux anymore, not even newly created ones (even after uninstalling the packages I initially added for "flutter_secure_storage").

### Actual results

"flutter run" fails with the error ```"undefined symbol: g_once_init_enter_pointer``` and then
```Error waiting for a debug connection: The log reader stopped unexpectedly, or never started.```
```Error launching application on Linux.```

From googling I believe the symbol should be part of GLib 2.80: https://docs.gtk.org/glib/type_func.Once.init_enter_pointer.html
According to the logs that is used:
```-- Checking for module 'glib-2.0'```
```--   Found glib-2.0, version 2.80.0```
I have tried several things like downgrading the GTK package, but with not success. Any pointers what else I could try?

### Logs

[Log flutter create and run with Flutter 3.19.3](https://gist.github.com/czaefferer/94666f934d2e872546da537daa931c2c)

[Log flutter run with Flutter master](https://gist.github.com/czaefferer/4440e767396fa8c1f3a285fc0d8b4d49)

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

cza@cza-laptop:/tmp/counter_test$ flutter doctor -v
[✓] Flutter (Channel stable, 3.19.3, on Ubuntu 22.04.4 LTS 5.15.0-100-generic, locale en_US.UTF-8)
    • Flutter version 3.19.3 on channel stable at /home/cza/.puro/envs/v3_19_3/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ba39319843 (12 days ago), 2024-03-07 15:22:21 -0600
    • Engine revision 2e4ba9c6fb
    • Dart version 3.3.1
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/cza/Android/Sdk
    • Platform android-34, build-tools 30.0.3
    • Java binary at: /opt/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • Homebrew clang version 17.0.6
    • cmake version 3.28.3
    • ninja version 1.11.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2023.1)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 77.1.2
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.87.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.84.0

[✓] Connected device (3 available)
    • SM N981B (mobile) • RFCN806888F • android-arm64  • Android 13 (API 33)
    • Linux (desktop)   • linux       • linux-x64      • Ubuntu 22.04.4 LTS 5.15.0-100-generic
    • Chrome (web)      • chrome      • web-javascript • Chromium 122.0.6261.128 snap

[✓] Network resources
    • All expected network resources are available.

• No issues found!
cza@cza-laptop:/tmp/counter_test$ 


</details>

**Comment by huycozy** [comment]
Hi @czaefferer 
I checked this issue on my Linux machine, also using Ubuntu distro but can't reproduce this. 

First, I add `flutter_secure_storage` package to a new Flutter project. Next, I install these libraries to the system: `libsecret-1-dev libjsoncpp-dev libsecret-1-0` (with apt). Then I run the project with Linux target app, the app can run normally.

I'm unsure if there are incompatible library versions between brew and apt. Could you try re-installing these libraries with apt and retry running Flutter app?

**Comment by czaefferer** [comment]
Hi @huycozy,
thank you very much, reinstalling the packages via apt didn't change the error, but the hint about possibly incompatible versions between brew and apt did help. I have uninstalled a bunch of libraries from brew (where the ones relevant for Flutter are also installed via apt), and now Linux as a target is working again.

This were the libraries I uninstalled, in case this helps anyone else (first four are the ones I wanted to get rid of, the rest were just dependent on them):
```brew uninstall cmake ninja pkg-config gtk+3 freeglut gobject-introspection libepoxy libpthread-stubs libvdpau libxinerama libxtst libxxf86vm meson at-spi2-core libxkbcommon mesa mesa-glu xinput xorg-server```

Notably, ```cmake``` was downgraded from  3.28.3 to 3.22.1 and ```ninja``` from 1.11.1 to 1.10.1 according to ```flutter doctor```, so I suspect one of those as the culprit.
